tests/bluetooth/audio/codec/src/main.c

```diff
@@ -772,6 +1025,93 @@ ZTEST(audio_codec_test_suite, test_bt_audio_codec_cap_set_max_codec_frames_per_s
 }
 
+ZTEST(audio_codec_test_suite, test_bt_audio_codec_cap_meta_get_val)
+{
+	struct bt_audio_codec_cap codec_cap =
+		BT_AUDIO_CODEC_CAP(BT_HCI_CODING_FORMAT_LC3, 0x0000, 0x0000, {},
+				   {BT_AUDIO_CODEC_DATA(BT_AUDIO_METADATA_TYPE_PARENTAL_RATING,
+							BT_AUDIO_PARENTAL_RATING_AGE_10_OR_ABOVE)});
+	const uint8_t expected_data = BT_AUDIO_PARENTAL_RATING_AGE_10_OR_ABOVE;
+	const uint8_t *data;
+	int ret;
+
+	ret = bt_audio_codec_cap_meta_get_val(&codec_cap, BT_AUDIO_METADATA_TYPE_PARENTAL_RATING,
+					      &data);
+	zassert_equal(ret, sizeof(expected_data), "Unexpected return value %d", ret);
+	zassert_equal(data[0], expected_data, "Unexpected return value %d", ret);
+}
+
+ZTEST(audio_codec_test_suite, test_bt_audio_codec_cap_meta_set_val)
+{
+	struct bt_audio_codec_cap codec_cap =
+		BT_AUDIO_CODEC_CAP(BT_HCI_CODING_FORMAT_LC3, 0x0000, 0x0000, {},
+				   {BT_AUDIO_CODEC_DATA(BT_AUDIO_METADATA_TYPE_PARENTAL_RATING,
+							BT_AUDIO_PARENTAL_RATING_AGE_10_OR_ABOVE)});
+	const uint8_t new_expected_data = BT_AUDIO_PARENTAL_RATING_AGE_13_OR_ABOVE;
+	const uint8_t expected_data = BT_AUDIO_PARENTAL_RATING_AGE_10_OR_ABOVE;
+	const uint8_t *data;
+	int ret;
+
+	ret = bt_audio_codec_cap_meta_get_val(&codec_cap, BT_AUDIO_METADATA_TYPE_PARENTAL_RATING,
+					      &data);
+	zassert_equal(ret, sizeof(expected_data), "Unexpected return value %d", ret);
+	zassert_equal(data[0], expected_data, "Unexpected return value %d", ret);
+
+	ret = bt_audio_codec_cap_meta_set_val(&codec_cap, BT_AUDIO_METADATA_TYPE_PARENTAL_RATING,
+					      &new_expected_data, sizeof(new_expected_data));
+	zassert_true(ret > 0, "Unexpected return value %d", ret);
+
+	ret = bt_audio_codec_cap_meta_get_val(&codec_cap, BT_AUDIO_METADATA_TYPE_PARENTAL_RATING,
+					      &data);
+	zassert_equal(ret, sizeof(new_expected_data), "Unexpected return value %d", ret);
+	zassert_equal(data[0], new_expected_data, "Unexpected data value %u", data[0]);
+}
+
+ZTEST(audio_codec_test_suite, test_bt_audio_codec_cap_meta_set_val_new)
+{
+	struct bt_audio_codec_cap codec_cap =
+		BT_AUDIO_CODEC_CAP(BT_HCI_CODING_FORMAT_LC3, 0x0000, 0x0000, {}, {});
+	const uint8_t new_expected_data = BT_AUDIO_PARENTAL_RATING_AGE_13_OR_ABOVE;
+	const uint8_t *data;
+	int ret;
+
+	ret = bt_audio_codec_cap_meta_get_val(&codec_cap, BT_AUDIO_METADATA_TYPE_PARENTAL_RATING,
+					      &data);
+	zassert_equal(ret, -ENODATA, "Unexpected return value %d", ret);
+
+	ret = bt_audio_codec_cap_meta_set_val(&codec_cap, BT_AUDIO_METADATA_TYPE_PARENTAL_RATING,
+					      &new_expected_data, sizeof(new_expected_data));
+	zassert_true(ret > 0, "Unexpected return value %d", ret);
+
+	ret = bt_audio_codec_cap_meta_get_val(&codec_cap, BT_AUDIO_METADATA_TYPE_PARENTAL_RATING,
+					      &data);
+	zassert_equal(ret, sizeof(new_expected_data), "Unexpected return value %d", ret);
+	zassert_equal(data[0], new_expected_data, "Unexpected return value %d", ret);
+}
+
+ZTEST(audio_codec_test_suite, test_bt_audio_codec_cap_meta_unset_val)
```

Comment:
IMO it would be nice to have one of the unset_val tests to test where multiple values have been set, and then one  of the middle values get unset, so that we properly verify the looping in the ltv_unset_val function

Reply:
For the record the extra test was added in commit 000a68ff60c2fe405e10eb8e7708f56df9a02b2b

Reply:
> For the record the extra test was added in commit [000a68f](https://github.com/zephyrproject-rtos/zephyr/commit/000a68ff60c2fe405e10eb8e7708f56df9a02b2b)

Incorrect: The test added by that is testing something else (the `set` functions). The test for the `unset` has been added to the commit in this PR.